src/finch/tensor.py

```diff
@@ -139,13 +164,23 @@ def preprocess_order(
 
         return permutation
 
-    def get_order(self, zero_indexing=True) -> tuple[int, ...]:
+    @classmethod
+    def get_lvl_ndim(cls, lvl: JuliaObj) -> int:
+        ndim = 0
+        for _ in range(np.MAXDIMS):
+            ndim += 1
+            lvl = lvl.lvl
```

Comment:
Before this line, I'd add a `jl.isa` call (and raise on invalidity) to make sure the nesting is done right -- Or is that already handled Julia-side?

Reply:
I tried in Julia repl and it looks that invalid structure can't be composed.